lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -671,7 +679,20 @@ def _on_config_changed(self, event: ConfigChangedEvent):  # noqa C901
                 RelationJoinedEvent(event.handle, PeerRelationName, self.app, self.unit)
             )
 
-        previous_deployment_desc = self.opensearch_peer_cm.deployment_desc()
+        perf_profile_needs_restart = None
+        if (
+            (previous_deployment_desc := self.opensearch_peer_cm.deployment_desc())
+            and previous_deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            and not self.upgrade_in_progress
+        ):
+            perf_profile_needs_restart = self.performance_profile.apply(
+                self.config.get(PERFORMANCE_PROFILE)
+            )
+        else:
```

Comment:
can you explain why we always defer if the current app is not `MAIN_ORCHESTRATOR` ?